Src/Support/Google.Apis.Auth/OAuth2/CredentialFactory.cs

```diff
@@ -57,60 +71,79 @@ public static T FromFile<T>(string credentialPath)
     /// </summary>
     /// <typeparam name="T">The type of the credential to create.</typeparam>
     /// <param name="stream">The stream that contains the JSON credential data.</param>
     /// <returns>The created credential.</returns>
-    public static T FromStream<T>(Stream stream)
-    {
-        try
-        {
-            return FromJsonParameters<T>(NewtonsoftJsonSerializer.Instance.Deserialize<JsonCredentialParameters>(stream));
-        }
-        catch (Exception e)
-        {
-            throw new InvalidOperationException("Error deserializing JSON credential data.", e);
-        }
-    }
+    public static T FromStream<T>(Stream stream) =>
+        FromJsonParameters<T>(HandleDeserializationErrors(() =>
+            NewtonsoftJsonSerializer.Instance.Deserialize<JsonCredentialParameters>(stream)));
+
+    /// <summary>
+    /// Creates a credential of the specified type from a stream that contains JSON credential data.
+    /// </summary>
+    /// <param name="stream">The stream that contains the JSON credential data.</param>
+    /// <param name="credentialType">The type of credential to be loaded. Valid strings can be found in <see cref="JsonCredentialParameters"/>.</param>
+    /// <returns>The created credential.</returns>
+    public static GoogleCredential FromStream(Stream stream, string credentialType) =>
+        FromJsonParameters(HandleDeserializationErrors(() =>
+            NewtonsoftJsonSerializer.Instance.Deserialize<JsonCredentialParameters>(stream)), credentialType);
 
     /// <summary>
     /// Creates a credential of the specified type from a string that contains JSON credential data.
     /// </summary>
     /// <typeparam name="T">The type of the credential to create.</typeparam>
     /// <param name="json">The string that contains the JSON credential data.</param>
     /// <returns>The created credential.</returns>
-    public static T FromJson<T>(string json)
-    {
-        try
-        {
-            return FromJsonParameters<T>(NewtonsoftJsonSerializer.Instance.Deserialize<JsonCredentialParameters>(json));
-        }
-        catch (Exception e)
-        {
-            throw new InvalidOperationException("Error deserializing JSON credential data.", e);
-        }
-    }
+    public static T FromJson<T>(string json) =>
+        FromJsonParameters<T>(HandleDeserializationErrors(() =>
+            NewtonsoftJsonSerializer.Instance.Deserialize<JsonCredentialParameters>(json)));
+
+    /// <summary>
+    /// Creates a credential of the specified type from a string that contains JSON credential data.
+    /// </summary>
+    /// <param name="json">The string that contains the JSON credential data.</param>
+    /// <param name="credentialType">The type of credential to be loaded.</param>
+    /// <returns>The created credential.</returns>
+    public static GoogleCredential FromJson(string json, string credentialType) =>
```

Comment:
We don't need these at the moment. We should remove them and just add them if there's a user need.